tasks/03_pages/scraper.py

```diff
@@ -0,0 +1,170 @@
+                       case_number: str,
+                       error: bool,
+                       raw_text: str):
+    
+    # if loading extant .html page files, get case_id from cases table
+    if case_id is None:
+        try:
+            with sql.db_cnx() as cnx:
+                c = cnx.cursor()
+                c.execute(f"""
+                        SELECT t1.id
+                        FROM cases t1
+                        WHERE t1.case_number = '{case_number}';
+                        """)
+                case_id = c.fetchone()
+                c.close()
+                cnx.close()
+        except Exception as e:
+            print(f'Error with case_id as None: {e}')
+
+    # insert relevant info to pages table in the db
+    try:
+        if db_config.db_type == 'sqlite':
+                query = '''INSERT INTO pages
+                            (case_id, case_number, error, raw_text)
+                            VALUES (?, ?, ?, ?)
+                        '''
+        elif db_config.db_type == 'postgresql':
+            query = """INSERT INTO pages (case_id, case_number, error, raw_text) 
+                            VALUES (%s, %s, %s, %s);
+                    """
+                
+        with sql.db_cnx() as cnx:
+            c = cnx.cursor()
+            c.execute(query, (int(case_id), case_number, error, raw_text))        
+            
+    except Exception as e:
+        print(f'Error adding page to {db_config.pages} table: {e}')
+    finally:
+        c.close()
+        cnx.close()
+
+
+def threaded_scraper(case_and_case_number: tuple):
+    """
+    
+    """
+    # scraper doesn't necessarily need to be run in threads
+    
+    # get some useful variables from the inputs
+    case_id, case_number = case_and_case_number
+    output_path = Path(paths.pages/f"{case_id}_{case_number}.html")
+    
+    # keep track of time
+    t1 = time.time()    
+
+
+    # try to scrape a page. 
+    try:
+        case_response = requests.get(f'https://www.nlrb.gov/case/{case_number}')
+        case_response.raise_for_status()
+    
+    # if you get an error, log it, add error to `pages` table, return out of scraper
+    except requests.exceptions.HTTPError as e:
+        print(f'Error: {e}, case: {case_number}')
+        logging.warning(f'{case_id}, {case_number}, fetch error: {e}')
+        
+        add_to_pages_table(case_id=case_id, 
+                           case_number=case_number,
+                           error=True,
+                           raw_text=False)
+        return
+    
+    soup = bs(case_response.text, 'lxml')
+    if soup.article is None:
+        print(f'Soup.article is none for: {case_number}')
+        logging.warning(f'{case_id}, {case_number}, write error:{soup.article.prettify()}')
+        add_to_pages_table(case_id=case_id,
+                               case_number=case_number,
+                               error=True,
+                               raw_text=False)
+        return
```

Comment:
instead of returning here, you might want to do something like:
```
if soup.article is None:
   <log error>
else:
    page = soup.article.prettify()
    try:
        <add page HTML to pages table>
    except Exception as e:
         <handle exception>
```